### PR TITLE
[WEB-4042] Add searchDataId Meganav prop

### PR DESCRIPTION
**Pull request by matt423**
## Jira Ticket Link / Motivation

WEB-4042

## Summary of changes

* Add `searchDataId` to `Meganav` props
* If `searchDataId` present then use that value for the `data-id` of search elements(input and button)
  * Allows for over-riding the components current onClick behaviour 
* Only add/remove search event listeners if they are present

## How do you manually test this?

**Storybook**

* [Added story component](https://ably.github.io/ably-ui/?path=/story/js-components-meganav--data-search-id)
  * Verify that `data-id` is changed when `searchDataId` is present in the props
  * Side affect that the `onclick` handlers aren't fired

## Release Notes

- **New Features**
	- Introduced a new optional property `searchDataId` across several components, enhancing the search functionality for signed-in users.
	- Added a new story for the `Meganav` component to showcase its functionality with the `searchDataId`.

- **Bug Fixes**
	- Improved robustness of event listeners in the `MeganavSearchSuggestions` component to prevent errors from null references.

- **Chores**
	- Updated the project version from `14.7.1` to `14.7.4`.

### CODE REVIEW

File: package.json
Comment (jamiehenson) [inline package.json:3]: `14.7.2` has been released now, so we'll need to target `14.7.3` here (maybe coordinate with @aralovelace to see what else that release could have, or whether this goes out by itself).

Also we should avoid dev versions in `package.json`

File: src/core/MeganavItemsMobile.tsx
Comment (jamiehenson) [inline src/core/MeganavItemsMobile.tsx:106]: `searchDataId ?? "meganav-mobile-search-input"` ?

**Review by jamiehenson — COMMENTED** [review]
Looks good to me and stamping in principle, but will defer to someone else to approve the PR when the version numbering is fixed

File: package.json
Comment (kennethkalmer) [reply package.json:3]: I published `14.7.3` already, so this should be part of `14.7.4`

Comment (matt423) [reply package.json:3]: Was mainly just a placeholder for the review app, will bump to `14.7.4`

**Review by kennethkalmer — APPROVED** [review]

**Pull request merged**
